### PR TITLE
Introduce a Prompting command :wink:

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Add feature to prompt for emoji to insert either in editor or in terminal 💻 
This comes in endy notably from writing commits from the commandline in the embedded terminal 😉 

Capture d'écran 2019-03-29 08 50 58

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,66 @@
...
+        },
+        {
+          "command": "emojisense.quickEmojitextTerminal",
+          "when": "panelFocus && activeViewlet == 'workbench.panel.terminal'"
+        }
+      ]
+    },
+    "keybindings": [
+      {
+        "mac": "cmd+[Period]",
```
Comment (mattbierner) [inline package.json]: This keybinding is already used to trigger code actions. Any ideas for a different one that does not conflict with VS Code's built-in keybindings?

File: src/quickEmoji.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	const items: QuickPickItem[] = Array.from(emoji.emojis).map(
+		emoji => ({label: emoji.name, description: emoji.emoji})
+	);
+	return async function () {
+		const terminal = window.activeTerminal;
+		const editor = window.activeTextEditor;
+		const emoji = await window.showQuickPick(items);
+		if (!emoji) return;
+		const insert = property === 'name' ? `:${emoji.label}:`: emoji.description || '';
+		if (terminal && destination === 'terminal') terminal.sendText(insert, false);
```
Comment (mattbierner) [inline src/quickEmoji.ts]: nit: add braces and line breaks

File: README.md [hunk 3]
```diff
@@ -10,6 +10,7 @@ Inspired by the [Atom autocomplete+ emojis suggestions plugin][atom].
 - Insert emoji markup by typing `::`
 - Enable and control emoji completion settings per language
 - See emoji previews of `:smile:` style markup inline
+- Provide an emoji picker to select and insert your emoji. (notably in the terminal) :wink:
```
Comment (mattbierner) [inline README.md]: Make sure to document how to use this feature and whatever the contributed keybindings end up being

File: package.json [hunk 4]
```diff
@@ -28,6 +28,66 @@
...
+        "key": "ctrl+[Period]",
+        "command": "emojisense.quickEmojiTerminal",
+        "when": "inputFocus"
```
Comment (mattbierner) [inline package.json]: Should this be `terminalFocus`?

File: package.json [hunk 5]
```diff
@@ -28,6 +28,66 @@
...
+        "mac": "shift+cmd+[Period]",
+        "key": "shift+ctrl+[Period]",
+        "command": "emojisense.quickEmojitextTerminal",
+        "when": "inputFocus"
+      },
+      {
+        "mac": "cmd+[Period]",
+        "key": "ctrl+[Period]",
+        "command": "emojisense.quickEmoji",
+        "when": "!terminalFocus"
```
Comment (mattbierner) [inline package.json]: `editorTextFocus` here I believe

File: package.json (hunk 1 above)
Comment (AdrieanKhisbe) [reply package.json]: Didnt know about that one! :)

Comment (AdrieanKhisbe) [reply package.json]: I tried out <kbd>cmd+[Minus]</kbd> on my computer, what do you think about that one?
(this is  <kbd>cmd+ )</kbd> on my french keyboard)

File: src/quickEmoji.ts (hunk 2 above)
Comment (AdrieanKhisbe) [reply src/quickEmoji.ts]: will do

**Comment by AdrieanKhisbe** [comment]
@mattbierner I take into account your reviews, and add some improvement.

Also rebased to fix conflict

**Comment by mattbierner** [comment]
Thanks! Will publish updated extension shortly

**Comment by AdrieanKhisbe** [comment]
@mattbierner Awesome!! 🙌  

=)

**Pull request merged**